docs/architecture/COSMWASM_POOLS.MD

```diff
@@ -0,0 +1,46 @@
+# CosmWasm Pools
```

Comment:
Do you mind adding a paragraph about semver and the general new version handling we arrived at with the cw pools? For example, "alloyed transmuter versions are expected to exist within the same major version i.e >=3.0.0 and <4.0.0"